### DISCUSSION TITLE
How to make a router link to a component in my page ?

### DISCUSSION THREAD

**Discussion by ghost** [quasarframework/quasar#10041] [General - Components / Directives / etc] [answered]
Hi, In my quasar app i have a Home page that has a number of components:

```
<template>
  <LandingSection />
  <AboutUs />
  <OurProgrammes />
  <WhyUs />
  <Video />
  <RegisterNow />
  <Footer />
</template>

<script>
import { defineAsyncComponent } from "vue";
export default {
  components: {
    LandingSection: defineAsyncComponent(() =>
      import("../components/Home/LandingSection.vue")
    ),

    AboutUs: defineAsyncComponent(() => import("../components/Home/AboutUs")),

    OurProgrammes: defineAsyncComponent(() =>
      import("../components/Home/OurProgrammes.vue")
    ),

    WhyUs: defineAsyncComponent(() => import("../components/Home/WhyUs.vue")),

    Video: defineAsyncComponent(() => import("../components/Home/Video.vue")),

    RegisterNow: defineAsyncComponent(() =>
      import("../components/Home/RegisterNow.vue")
    ),

    Footer: defineAsyncComponent(() => import("../components/Home/Footer.vue")),
  },
};
</script>

```

I also have a scrolling to elements using the `getScrollTarget()` and `setVerticalScrollPosition()` in my App.vue file like this:

```
<script>
import { scroll } from "quasar";
const { getScrollTarget, setVerticalScrollPosition } = scroll;
export default {
  methods: {
    scrollToElement(scrollTarget, currentLabel) {
      this.activeLabel = currentLabel;
      this.leftDrawerOpen = false;
      const el = document.getElementsByClassName(scrollTarget)[0];
      const target = getScrollTarget(el);
      const offset = el.offsetTop - 64;
      const duration = 1000;
      setVerticalScrollPosition(target, offset, duration);
    },
  },
};
</script>

```

This is my routes,js file:

```
const routes = [
  {
    name: "Home",
    path: "/",
    component: () => import("pages/Home.vue"),
  },
  {
    name: "RegisterNow",
    path: "/register-now",
    component: () => import("pages/RegisterNow.vue"),
  },
  // Always leave this as last one,
  // but you can also remove it
  {
    path: "/:catchAll(.*)*",
    component: () => import("pages/Error404.vue"),
  },
];

export default routes;

```

Now i want to make a router link to the `RegisterNow` component (or any component) in my home page so that anyone can get that link and go directly to that part of the page ?

Like here in quasar docs there are links for every part of the page and when you click on the title you copy the link to that part of the page
(https://quasar.dev/quasar-cli/lazy-loading#lazy-load-router-pages)

Also here in vue router page (https://router.vuejs.org/guide/) if you scroll at the bottom of the page you will find that the url changes to (https://router.vuejs.org/guide/#javascript) but if you scroll up again the url goes back to (https://router.vuejs.org/guide/#html), How is this dunamic url change happens ?

**Comment by hawkeye64** [comment, marked as answer]
This is basic HTML. Where ever you want it to "jump" to a particular position on a page, you have to give a component an `id` (ex: `<h2 id="jump-to-here" ...`). Then in the URL, `https://my-site.com#jump-to-here` will load the page and then "jump" to that position.

https://blog.hubspot.com/marketing/jump-link-same-page?toc-variant-b=

**Reply by ghost** [reply]
I already tried that, my home route is `http://localhost:8080/#/` or  `http://localhost:8080/#/home` if i add the id and go to `http://localhost:8080/#/#register-now` nothing changes. Notice that i am using webhash

**Reply by hawkeye64** [reply]
ah, sorry. Using hash only works in history mode, not hash mode (obviously, the router is already using it in hash mode and would conflict with your coding).

**Reply by hawkeye64** [reply]
Note, with history mode, if using a backend server, you may have to make other backend considerations.

Go here: https://quasar.dev/quasar-cli/quasar-conf-js#property-build
Search (ctrl-f) for: vueRouterMode

**Reply by ghost** [reply]
I tried using history mode before and serve the vue app using an express server, but i had conflict in the routes and instead of hitting the routes of the vue app, i hit the routes of the backend express server. So i used a separate nginx server to serve the vue app and another express server for serving the data.

Is that the right thing to do ? and is there a way to serve the vue app directly from the express server without problems ?

**Reply by hawkeye64** [reply]
If you are using express, then as your last route, you can do this:

```js
  // should always be last - the catch-all
  app.get('*', function (req, res) {
    //   console.log(`URL not found!: '${req.url}'`)
    //   console.log('Serving index.html')
    res.sendFile('/index.html', { root: distPath })
  })
```

**Reply by ghost** [reply]
Thank you so much man, that helped a lot

**Reply by ghost** [reply]
> If you are using express, then as your last route, you can do this:
> 
> ```js
>   // should always be last - the catch-all
>   app.get('*', function (req, res) {
>     //   console.log(`URL not found!: '${req.url}'`)
>     //   console.log('Serving index.html')
>     res.sendFile('/index.html', { root: distPath })
>   })
> ```

What if i want to have more than one SPA, if i have a vue app in `public` folder for example and another one in `admin` folder, with the approach you used i can only render one app which will be the last route, how can i render more than one app ?
